### ISSUE TITLE
Rules implementation: Deteriorating Quality Rating of components

### ISSUE THREAD

**Issue by inqistor** [MegaMek/mekhq#1234]
Since MekHQ actually can trace Quality Rating of individual components, I suppose it wouldn't be hard to implement deteriorating rules from StratOps. I don't think anyone sane used those rules on paper, but since here computer takes care of all calculations, why not?

StratOps page 191 - SALVAGE (from battlefield, after battle):
```
If the Quality Rating of individual parts is being
tracked, the rating of any salvaged part automatically drops
one level. Any component that drops to Salvage status (Quality
Rating A) suffers the penalties outlined on the Salvage Quality
Equipment Table (see p. 192). Any component reduced past
Salvage status is treated as destroyed.
```
Page 188 - INSTALLING A REFIT KIT:
```
Failure doubles the time and also reduces the Quality Rating of
the unit by one level. If players are tracking the Quality Rating of
individual components, then all the items included in the refit kit
have their Quality Rating reduced by one.
```
And page 177 - MECH HULL BREACH:
```
A ’Mech that has suffered a hull breach will drop to the next
Quality Rating. If players are tracking the quality of individual
components, everything in the breached location is reduced
to the next Quality Rating.
```
Basically in all those instances, Quality Rating of components in location drops by one category.
